From c30d7af024bcdb8cda7beb1c66826af30397b09d Mon Sep 17 00:00:00 2001
From: ozeng <oak.zeng@amd.com>
Date: Mon, 6 Mar 2017 14:37:46 -0600
Subject: [PATCH 1976/2094] drm/amdkfd: Changed SPI priority mapping

SPI priority 2 and 3 are reserved for trap handler context
save. So only map SPI priority 0 and 1 for application

Change-Id: Id95831af4fc0b2b818ccfe6cc6da681c1ecfa843
Signed-off-by: Oak Zeng <Oak.Zeng@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager.c | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager.c b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager.c
index 5a5dd2a..0980995 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager.c
@@ -43,24 +43,26 @@ int pipe_priority_map[] = {
 	KFD_PIPE_PRIORITY_CS_HIGH
 };
 
-/* Mapping queue priority to SPI priority, indexed by queue priority */
+/* Mapping queue priority to SPI priority, indexed by queue priority
+ * SPI priority 2 and 3 are reserved for trap handler context save
+ */
 int spi_priority_map[] = {
 	KFD_SPI_PRIORITY_EXTRA_LOW,
 	KFD_SPI_PRIORITY_EXTRA_LOW,
 	KFD_SPI_PRIORITY_EXTRA_LOW,
 	KFD_SPI_PRIORITY_EXTRA_LOW,
+	KFD_SPI_PRIORITY_EXTRA_LOW,
+	KFD_SPI_PRIORITY_EXTRA_LOW,
+	KFD_SPI_PRIORITY_EXTRA_LOW,
+	KFD_SPI_PRIORITY_EXTRA_LOW,
+	KFD_SPI_PRIORITY_LOW,
+	KFD_SPI_PRIORITY_LOW,
+	KFD_SPI_PRIORITY_LOW,
 	KFD_SPI_PRIORITY_LOW,
 	KFD_SPI_PRIORITY_LOW,
 	KFD_SPI_PRIORITY_LOW,
 	KFD_SPI_PRIORITY_LOW,
-	KFD_SPI_PRIORITY_MEDIUM,
-	KFD_SPI_PRIORITY_MEDIUM,
-	KFD_SPI_PRIORITY_MEDIUM,
-	KFD_SPI_PRIORITY_MEDIUM,
-	KFD_SPI_PRIORITY_HIGH,
-	KFD_SPI_PRIORITY_HIGH,
-	KFD_SPI_PRIORITY_HIGH,
-	KFD_SPI_PRIORITY_HIGH
+	KFD_SPI_PRIORITY_LOW
 };
 
 struct mqd_manager *mqd_manager_init(enum KFD_MQD_TYPE type,
-- 
2.7.4

